Add bitcoin network slides (#28)
cifunibas/Bitcoin-Blockchain-Cryptoassets: slides/bitcoin_network.tex

% Choose one to switch between slides and handout
%\documentclass[]{beamer}
\documentclass[handout]{beamer}

% Video Meta Data
\title{Bitcoin, Blockchain and Cryptoassets}
\subtitle{The Bitcoin Network}
\author{Prof. Dr. Fabian Schär}
\institute{University of Basel}

% Config File
\input{../config/config.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\thispagestyle{empty}
\begin{frame}[noframenumbering]
	\titlepage
\end{frame}

%%%
\begin{frame}{Bitcoin is a Peer-to-Peer Network}
	\centering
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
		\input{../assets/figures/p2p}

> 		\color{focus}
> 		\node (text) at (-0.5,3.5) { Nodes };
> 		\draw[->, thick]	(text.south) -- (agenta.west);
> 		\draw[->, thick]	(text.south) -- (agentb.north west);

Reviewer: ```suggestion
```

	\end{tikzpicture}
\end{frame}
%%%

%%%
\begin{frame}{Joining the Network}
	\begin{center}
		\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
			\input{../assets/figures/join_network}
		\end{tikzpicture}
	\end{center}

	\vspace{1cm}
	

> 	After the initial handshake, we request a list of addresses and connect to more peers.

Reviewer: ```suggestion
	After the initial handshake, we request a list of IP addresses and connect to additional peers.
```

\end{frame}
%%%

%%%
\begin{frame}{Functionality of a Full Node}
	\centering
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
		\input{../assets/figures/node_functions}
	\end{tikzpicture}
\end{frame}
%%%

%%%
\begin{frame}{Verify New Blocks}
	
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
		\input{../assets/figures/full_node_functions}
	\end{tikzpicture}
\end{frame}
%%%

%%%
\begin{frame}{Distribute Verified Blocks}
	
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
		\input{../assets/figures/full_node_functions2}
	\end{tikzpicture}
\end{frame}
%%%

%%%
\begin{frame}{Wallet Functionality}
	\centering
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]

> 		\input{../assets/figures/wallet_functions}

Reviewer: Encrypted Key Store. Also please remove the term security layer.

	\end{tikzpicture}
\end{frame}
%%%

\begin{frame}{Mining Functionality}
	\centering
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]

> 		\input{../assets/figures/mining_functions}

Reviewer: Create block -> Assemble Blocks
Also, please change order, i.e. assemble blocks in the left and compute on the right.

	\end{tikzpicture}
\end{frame}

%%%
\begin{frame}{Node Distribution}
	\begin{figure}[t]
		\includegraphics[width = 0.9\textwidth]{../assets/images/node_distribution}
		Source: \link \url{https://bitnodes.io/}, March 2021
	\end{figure}

	\begin{footnotesize}
		\begin{itemize}
			\item $\sim$50\% of nodes in Europe
			\item $\sim$20\% of nodes in the USA
		\end{itemize}
	\end{footnotesize}
	
\end{frame}
%%%

%%%
\begin{frame}{Historic Number of Full Nodes}
	\begin{description}
		\item[\textbf{Dec 2013:}] $\sim$15500 full nodes
		\item[\textbf{Dec 2015:}] $\sim$5500 full nodes
		\item[\textbf{Mar 2017:}] $\sim$6500 full nodes
		\item[\textbf{Mar 2019:}] 10362 full nodes
		\item[\textbf{Mar 2021:}] 10179 full nodes
	\end{description}	
		
	\vspace{1.5 cm}
		
	Reasons for decline?
	
\end{frame}
%%%

> %%%
> \begin{frame}{Network Partitions}
> 	\centering
> 	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
> 		\input{../assets/figures/partitioned_network}
> 	\end{tikzpicture}
> \end{frame}
> %%%

Reviewer: ```suggestion
```

%%%
\begin{frame}{Extended Network}
	\centering
	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
		\input{../assets/figures/extended_network}
	\end{tikzpicture}
\end{frame}
%%%

%%%
\begin{frame}{Simplified Payment Verification}

> 

Reviewer: Less Centralized options:
* Simplified Payment Verification (SPV).
* Running full node at home and connect to it with mobile devices.

\end{frame}
%%%

%%%

> \begin{frame}{Pool Mining}

Reviewer: Move up.

	\centering
	\begin{figure}[t]
		\includegraphics[height = 0.8\textheight]{../assets/images/mining_pools}
		
		Source: \link \url{https://blockchain.info/pools}, March 2021
	\end{figure}
\end{frame}
%%%

> 
> %%%
> \begin{frame}{Recommended Literature}
> 	
> \end{frame}
> %%%

Reviewer: ```suggestion
```
